File: List/RemoveDuplicatesFromUnsortedList/RemoveDuplicatesFromUnsortedList.py
"""
class ListNode(object):
    def __init__(self, val, next = none):
        self.val = val
        self.next = next
"""

class Solution:
    def rmDupFromUnsortedList(self, head):
        if head is None:
            return None
            
        curr = head
        while curr is not none:
            inner = curr
            while inner.next is not None:
                if inner.next.val == curr.val:
                    inner.next = inner.next.next
                else:
                    inner = inner.next
            curr = curr.next
        return head

File: List/RemoveDuplicatesFromUnsortedList/RemoveDuplicatesFromUnsortedList.java
/*
class ListNode {
	int val;
	ListNode next;
	
	ListNode(int val) {
		this.val = val;
		this.next = null;
	}
};
*/

class Solution {
	public ListNode rmDupFromUnsortedList(ListNode head) {
		if(head == null)
			return null;
		
		ListNode curr = head;
		while(!curr) {
			ListNode inner = NULL;
			while(inner.next != NULL  && inner.next.next != NULL) {
				if(inner.next.val == curr.val) {
					inner.next = inner.next.next
				} else
					inner = inner.next;
			}
			curr = curr.next;
		}
		return head;
	}
}